src/Fuzz.elm

```diff
@@ -357,13 +365,35 @@ listShrinkHelp listOfTrees =
         root =
             List.map RoseTree.root listOfTrees
 
+        dropFirstHalf : List (RoseTree a) -> RoseTree (List a)
+        dropFirstHalf list_ =
+            List.drop (List.length list_ // 2) list_
+                |> listShrinkRecurse
+
+        dropSecondHalf : List (RoseTree a) -> RoseTree (List a)
+        dropSecondHalf list_ =
+            List.take (List.length list_ // 2) list_
+                |> listShrinkRecurse
+
+        halved : LazyList (RoseTree (List a))
+        halved =
+            if n >= 4 then
+                Lazy.lazy <|
+                    \_ ->
+                        Lazy.List.fromList [ dropFirstHalf listOfTrees, dropSecondHalf listOfTrees ]
+                            |> Lazy.force
+            else
+                -- For lists of three elements or less, halving and removing a single element is often going to be the same thing.
+                -- To prevent us from attempting the same shrunken value twice, we're disabling the halving strategy for these small lists.
```

Comment:
Yeah, sure, but that's fundamentally an optimization; the fuzzer will do (more) duplicate work if you leave that check out, but it'll still be correct. My reasoning is that if it's an performance optimization, it should look like one, so you can glance over it quickly if you just want to know what this function does.